.github/workflows/build_frontend-ng.yml

```
@@ -0,0 +1,39 @@
+name: Build and Deploy Frontend-NG
+
+on:
+  push:
+    branches:
+      - main # Or your main development branch
+    paths:
+      - 'timesketch/frontend-ng/**'
+      - '!timesketch/frontend-ng/dist/**' # Exclude changes in the dist folder
+  workflow_dispatch:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18' # Or your preferred Node.js version
```

I think we are running on Node 20 by now.

```suggestion
          node-version: '20'
```

---

ack